remote-wallet/src/trezor.rs

```diff
@@ -0,0 +1,265 @@
+use {
+    crate::remote_wallet::{
+        RemoteWallet, RemoteWalletError, RemoteWalletInfo, RemoteWalletManager,
+    },
+    bitcoin::bip32,
+    console::Emoji,
+    semver::Version as FirmwareVersion,
+    solana_sdk::{derivation_path::DerivationPath, pubkey::Pubkey, signature::Signature},
+    std::{cell::RefCell, fmt, rc::Rc, str::FromStr},
+    trezor_client::{
+        client::common::handle_interaction,
+        protos::{SolanaGetPublicKey, SolanaPublicKey, SolanaSignTx, SolanaTxSignature},
+        utils::convert_path,
+        Trezor,
+    },
+};
+
+static CHECK_MARK: Emoji = Emoji("✅ ", "");
+
+/// Trezor Wallet device
+pub struct TrezorWallet {
+    pub trezor_client: Option<Rc<RefCell<Trezor>>>,
+    pub pretty_path: String,
+}
+
+impl fmt::Debug for TrezorWallet {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "trezor_client")
+    }
+}
+
+impl TrezorWallet {
+    pub fn new(trezor_client: Option<Rc<RefCell<Trezor>>>, pretty_path: String) -> Self {
+        Self {
+            trezor_client,
+            pretty_path,
+        }
+    }
+
+    pub fn get_trezor_firmware_version(&self) -> Result<FirmwareVersion, RemoteWalletError> {
+        match &self.trezor_client {
+            Some(trezor_client) => {
+                let features = trezor_client
+                    .borrow_mut()
+                    .features()
+                    .expect("no features")
```

Comment:
While I'm not sure how often a device would fail to return features, let's handle this gracefully with an Error instead of panicking.
Same comment for the next 2 methods.